pandas/core/groupby/groupby.py

```diff
@@ -1146,14 +1145,7 @@ def get_group(self, name, obj=None) -> DataFrame | Series:
             indexer = inds if self.axis == 0 else (slice(None), inds)
             return self._selected_obj.iloc[indexer]
         else:
-            warnings.warn(
-                "obj is deprecated and will be removed in a future version. "
-                "Do ``df.iloc[gb.indices.get(name)]`` "
-                "instead of ``gb.get_group(name, obj=df)``.",
-                FutureWarning,
-                stacklevel=find_stack_level(),
-            )
-            return obj._take_with_is_copy(inds, axis=self.axis)
+            raise ValueError("cannot pass argument obj to get_group")
```

Comment:
Sorry if I missed any conversations on this but I am fairly certain you can just remove the `obj=None` parameter from the function definition and then remove this `if...else` block altogether

Reply:
Agreed

Reply:
thank you, I removed the parameter `obj` from the definition of `get_group`, the block `if...else` and corrected docstring.